dist/ethereum.js

```diff
@@ -2604,39 +2615,65 @@ var requestManager = function() {
         }
 
         // HTTP ASYNC (only when callback is given, and it a HttpProvidor)
-        if(typeof callback === 'function' && provider.name === 'HTTP'){
+        if(typeof callback === 'function' && provider.name === 'RPC over HTTP'){
             provider.send(payload, function(result, status){
 
-                if (!jsonrpc.isValidResponse(result)) {
-                    if(typeof result === 'object' && result.error && result.error.message) {
-                        console.error(result.error.message);
-                        callback(result.error);
-                    } else {
-                        callback(new Error({
-                            status: status,
-                            error: result,
-                            message: 'Bad Request'
-                        }));
-                    }
-                    return null;
+                if(!utils.isArray(data)) {
+                    callback(new Error({
+                        status: status,
+                        error: result,
+                        message: provider.name +' didn\'t repsond with an array of method responses'
+                    }));
+                    return;
```

Comment:
Because we are always sending batch request, so we expect arrays back at all times. If its not an array its already a wrong response. 

The return is necessary, as he would otherwise try to iterate over the non-array type. (the error before is not thrown, but passed to the callback)